src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -136,320 +168,428 @@ public String getReference() {
+
+							// use a CONNECT method to establish a socket connection to a third party, via the server
+							// being tested
+							boolean connectWorks = testConnect(this.getBaseMsg(), thirdpartyHost, thirdpartyPort, thirdPartyContentPattern);
+							if (connectWorks) {
+								evidence = "";
+								alertMessage = optionsmsg; // there is no connectmessage, since the HttpSender does not
+															// support CONNECT
+								raiseAlert = true;
+								description = Constant.messages.getString("ascanbeta.insecurehttpmethod.connect.exploitable.desc",
+										enabledmethod);
+								extraInfo = Constant.messages.getString("ascanbeta.insecurehttpmethod.connect.exploitable.extrainfo",
+										thirdpartyHost);
+							}
+						} else {
+							throw new Exception("Cannot verify unrecognised HTTP method '" + enabledmethod + "'");
 						}
+
+					} else {
+						// == Low threshold --> no need to verify it
+						evidence = enabledmethod;
+						alertMessage = optionsmsg;
+						raiseAlert = true;
+						description = Constant.messages.getString("ascanbeta.insecurehttpmethod.desc", enabledmethod);
+						extraInfo = Constant.messages.getString("ascanbeta.insecurehttpmethod.extrainfo", allowedmethods);
+					}
+
+					if (raiseAlert) {
+						if (log.isDebugEnabled()) {
+							log.debug("Raising alert for Insecure HTTP Method");
+						}
+						// bingo.
+						bingo(Alert.RISK_MEDIUM, Alert.CONFIDENCE_MEDIUM,
+								Constant.messages.getString("ascanbeta.insecurehttpmethod.detailed.name", insecureMethod), description,
+								null, // originalMessage.getRequestHeader().getURI().getURI(),
+								null, // parameter being attacked: none.
+								"", // attack
+								extraInfo, Constant.messages.getString("ascanbeta.insecurehttpmethod.soln"), evidence, // evidence,
+																														// highlighted in
+																														// the message
+								alertMessage);
+					}
+				} else {
+					if (log.isDebugEnabled()) {
+						log.debug("enabledmethod does not match insecureMethod: " + enabledmethod + "!=" + insecureMethod);
 					}
 				}
 			}
-		} catch (Exception e) {
-			log.error("Error scanning a Host for Insecure HTTP Methods: " + e.getMessage(), e);
 		}
-		
+	}
+
+	private HttpRequestBase getHttpMethodForString(String httpMethod, String urlStr) {
+
+		HttpRequestBase ret = null;
+
+		if (httpMethod.equals("POST")) {
+			ret = new HttpPost(urlStr);
+		} else if (httpMethod.equals("HEAD")) {
+			ret = new HttpHead(urlStr);
+		} else if (httpMethod.equals("PUT")) {
+			ret = new HttpPut(urlStr);
+		} else if (httpMethod.equals("DELETE")) {
+			ret = new HttpDelete(urlStr);
+		} else if (httpMethod.equals("TRACK")) {
+			ret = new HttpTrace(urlStr);
+		} else if (httpMethod.equals("TRACE")) {
+			ret = new HttpTrace(urlStr);
+		} else if (httpMethod.equals("OPTIONS")) {
+			ret = new HttpOptions(urlStr);
+		}
+
+		return ret;
+	}
+
+	private int tryHttpMethod(HttpRequestBase method) throws Exception {
```

Comment:
It should use the method `sendAndReceive` to send the messages, as is now it bypasses the core configurations (e.g. auth, proxy, mode...).